cloudinit/distros/__init__.py

```diff
@@ -795,6 +795,32 @@ def shutdown_command(self, *, mode, delay, message):
             args.append(message)
         return args
 
+    def manage_service(self, action, service):
+        init_cmd = self.init_cmd
+        if self.uses_systemd() or 'systemctl' in init_cmd:
+            init_cmd = ['systemctl']
+            cmds = {'stop': ['stop', service],
+                    'start': ['start', service],
+                    'enable': ['enable', service],
+                    'restart': ['restart', service],
+                    'reload': ['reload-or-restart', service],
+                    'try-reload': ['reload-or-try-restart', service],
+                    }
+        else:
+            cmds = {'stop': [service, 'stop'],
+                    'start': [service, 'start'],
+                    'enable': [service, 'start'],
+                    'restart': [service, 'restart'],
+                    'reload': [service, 'restart'],
+                    'try-reload': [service, 'restart'],
+                    }
+        cmd = list(init_cmd) + list(cmds[action])
+        try:
```

Comment:
I think it'd be better here to `return subp.subp(...)` and leave the exception handling up to the caller.

E.g., cc_fan currently logs a warning, but cc_ntp raises. 

We can add a docstring and specify that it may raise ProcessExecutionError